hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/OperationRunnerFactory.java

```diff
@@ -0,0 +1,10 @@
+package com.hazelcast.spi.impl.operationexecutor;
+
+public interface OperationRunnerFactory {
+
+    OperationRunner createPartitionRunner(int partitionId);
+
+    OperationRunner createGenericRunner();
+
+    OperationRunner createAdHocRunner();
```

Comment:
Good one. I'll add documentation.

There are 3 types of runners:
- partition runners; a runner responsible for processing all operations for a given partition
- generic runners: a runner responsible for processing generic operations for a given generic thread
- a ad hoc runner: a runner that is used to process generic operations, but it isn't done on a generic thread. For example in the system all kinds of operations are executed on the calling thread (e.g. partition/cluster internals). But.. you need to have an operation runner for them.. but the problem is that operation runners normally are not threadsafe; they have state like .e.g currentTask.

With the ad hoc runner this isn't the case.. it is fully immutable and can be shared between threads.